test/minio/1.7.0/questions.yaml

```diff
@@ -184,6 +184,28 @@ questions:
       $ref:
         - "definitions/certificate"
 
+  - variable: logsearchapi
+    label: Log Search API Configuration
+    description: Log Search API Configuration
+    group: Minio Configuration
+    schema:
+      type: dict
+      additional_attrs: true
```

Comment:
We should avoid using this as it means user can add additional attrs which will never be validated and will remain there always.
I suppose you used it because you have specified defaults in `ix_values` which we can change in that case and define them here instead?

Reply:
It was only added, because I remember having validation issues on TrueCharts with it.
Can you please explain what it does exactly? I couldn't find anything regarding this.

Reply:
So a dict can only have attributes which are explicitly listed in it but if you specify this, it means that users can specify any other attrs they like `a` or `xyz` or something else and it won't be validated by middleware because it's not defined in the schema..
You would probably be having issues with it because of how you specify defaults in `ix_values.yaml` which means that you should be changing that as it won't be good practice to allow users to specify `a/b/c` etc to a dict.

Reply:
> So a dict can only have attributes which are explicitly listed in it but if you specify this, it means that users can specify any other attrs they like `a` or `xyz` or something else and it won't be validated by middleware because it's not defined in the schema.. You would probably be having issues with it because of how you specify defaults in `ix_values.yaml` which means that you should be changing that as it won't be good practice to allow users to specify `a/b/c` etc to a dict.

Which makes it borderline impossible to write clean helm-charts.

You should be open to the standard helm practice:
Defaults stay in values.yaml (in this case ix_values.yaml) and the USER configuration stays in questions.yaml.

However, your statement that users can randomly add dicts is completely untrue and values added in ix_values.yaml will **not** stay "forever" either.

We're doing this for TrueCharts for absolute ages and it works flawlessly for 1.5 year by now (including changing values in ix_values.yaml), so i think i'm qualified to say you're wrong here ;-)

Reply:
Note: the second you prevent questions.yaml to require 5 rows to define 1 row of values.yaml defaults, we can talk about using it. But the current design is not usable and is completely non-standard.

I also want to note that this practice is **NOT** compatible with the common chart that is designed as agreed with @sonicaj

Reply:
> So a dict can only have attributes which are explicitly listed in it but if you specify this, it means that users can specify any other attrs they like `a` or `xyz` or something else and it won't be validated by middleware because it's not defined in the schema.. You would probably be having issues with it because of how you specify defaults in `ix_values.yaml` which means that you should be changing that as it won't be good practice to allow users to specify `a/b/c` etc to a dict.

Okay, I see, spoke with Waqar too and got few more questions solved too.